File: VB/WpfApplication1/CustomParameterTemplateSelector.vb
Imports DevExpress.Xpf.Printing.Parameters
Imports DevExpress.Xpf.Printing.Parameters.Models
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace WpfApplication1
    Friend Class CustomParameterTemplateSelector
        Inherits ParameterTemplateSelector

        Public Overrides Function SelectTemplate(ByVal item As Object, ByVal container As System.Windows.DependencyObject) As System.Windows.DataTemplate
            Dim parameterModel = TryCast(item, ParameterModel)
            If parameterModel.Name = "CategoryID" Then
                Return LookUpEditTemplate
            End If
            Return MyBase.SelectTemplate(item, container)
        End Function
    End Class
End Namespace

File: CS/WpfApplication1/CustomParameterTemplateSelector.cs
using DevExpress.Xpf.Printing.Parameters;
using DevExpress.Xpf.Printing.Parameters.Models;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WpfApplication1
{
    class CustomParameterTemplateSelector : ParameterTemplateSelector
    {
        public override System.Windows.DataTemplate SelectTemplate(object item, System.Windows.DependencyObject container)
        {
            var parameterModel = item as ParameterModel;
            if (parameterModel.Name == "CategoryID")
                return LookUpEditTemplate;
            return base.SelectTemplate(item, container);
        }
    }
}
